coordinator/src/main/java/org/apache/uniffle/coordinator/ApplicationManager.java

```diff
@@ -152,6 +155,50 @@ public void refreshAppId(String appId) {
     }
   }
 
+  public void refreshRemoteStorages(ClientConf dynamicClientConf) {
+    if (dynamicClientConf == null || MapUtils.isEmpty(dynamicClientConf.getRemoteStorageInfos())) {
+      LOG.info("Refresh remote storage with empty config");
+      for (String path : availableRemoteStorageInfo.keySet()) {
+        removePathFromCounter(path);
+      }
+      availableRemoteStorageInfo.clear();
+      return;
+    }
+
+    Map<String, RemoteStorageInfo> remoteStorageInfoMap = dynamicClientConf.getRemoteStorageInfos();
+    LOG.info("Refresh remote storage with {}", remoteStorageInfoMap);
+
+    for (Map.Entry<String, RemoteStorageInfo> entry : remoteStorageInfoMap.entrySet()) {
+      String path = entry.getKey();
+      RemoteStorageInfo rsInfo = entry.getValue();
+
+      if (!availableRemoteStorageInfo.containsKey(path)) {
+        remoteStoragePathRankValue.computeIfAbsent(
+            path,
+            key -> {
+              // refreshRemoteStorage is designed without multiple thread problem
+              // metrics shouldn't be added duplicated
+              addRemoteStorageMetrics(path);
+              return new RankValue(0);
+            });
+      }
+      availableRemoteStorageInfo.put(path, rsInfo);
+    }
+    // remove unused remote path if exist
+    List<String> unusedPath = Lists.newArrayList();
+    for (String existPath : availableRemoteStorageInfo.keySet()) {
+      if (!remoteStorageInfoMap.keySet().contains(existPath)) {
```

Comment:
also use `containsKey`? and can a similar refactoring to line 227 be done here as well?